docs/onboarding/all.must_read_checklist.md

```diff
@@ -0,0 +1,48 @@
+<!-- toc -->
```

Comment:
all.must_read_checklist.reference.md 

(according to https://diataxis.fr/)

Also this is specific of developer (since it doesn't cover setting up and so on).
So maybe all.dev_must_read...?

Your call